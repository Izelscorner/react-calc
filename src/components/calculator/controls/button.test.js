import React from 'react';
import { shallow } from 'enzyme';
import Button from './button';
import toJson from 'enzyme-to-json';

describe('<Button />', () => {
  let wrapper;
  let handleEvent;

  beforeAll(() => {
    handleEvent = jest.fn();
    wrapper = shallow(<Button handleEvent={handleEvent} value="1" />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the button and value', () => {
    const button = wrapper.find('button');
    expect(button).toHaveLength(1);
    expect(button.text()).toEqual('1');
  });

  it('calls handleEvent on button click', function () {
    const button = wrapper.find('button');
    expect(button).toHaveLength(1);
    button.simulate('click');
    expect(handleEvent).toHaveBeenCalledWith("1");
  });
});
