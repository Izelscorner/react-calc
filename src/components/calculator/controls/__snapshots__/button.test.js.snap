// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  1
</button>
`;
